screeps-game-api/src/objects/mod.rs

```diff
@@ -190,16 +190,22 @@ impl HasPosition for RoomPosition {
     }
 }
 
-/// All RoomObjects have positions.
-impl<T> HasPosition for T
-where
-    T: RoomObjectProperties,
-{
-    fn pos(&self) -> RoomPosition {
-        js_unwrap!(@{self.as_ref()}.pos)
+/// Trait covering all objects with an id.
+pub trait HasId : AsRef<Reference> {
```

Comment:
What would you think of making this an `unsafe trait` to match the others?

It has a default method does relying on the behavior of `AsRef<Reference>::as_ref` returning something specific (something with an id method) like the other unsafe traits, and unlike `HasPosition`.